hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngine.java

```diff
@@ -147,4 +151,26 @@
     Map<String, String> getClientStatistics();
 
     String getOwnerUuid(String clientUuid);
+
+    /**
+     * Locates the cluster member that has the provided client address and returns its member address,
+     * to be used for intra-cluster communication. This is required when clients deliver messages with
+     * designated target members, since clients may be unaware of the actual member address (when
+     * advanced network config is enabled).
+     * Throws a {@link com.hazelcast.spi.exception.TargetNotMemberException} when no member with the
+     * provided client address can be located.
+     *
+     * @param clientAddress the client address of the member
+     * @return              the member address of the member
+     */
+    Address memberAddressOf(Address clientAddress);
```

Comment:
There is no usages of this method, but I guess it will be used in some dependent PRs.